timemachine/md/minimizer.py

```diff
@@ -277,3 +278,83 @@ def equilibrate_host(
     ctxt.multiple_steps(np.linspace(0.0, 0.0, n_steps))
 
     return ctxt.get_x_t(), ctxt.get_box()
+
+
+def get_val_and_grad_fn(impls, box, lamb):
+    """
+    Convert impls, box, lamb into a function that only takes in coords.
+    """
+
+    def val_and_grad_fn(coords):
+        nrgs = []
+        grads = []
+        for impl in impls:
+            g, _, u = impl.execute(coords, box, lamb)
+            nrgs.append(u)
+            grads.append(g)
+        return np.sum(nrgs, axis=0), np.sum(grads, axis=0)
+
+    return val_and_grad_fn
+
+
+def local_minimize(x0, val_and_grad_fn, local_idxs):
+    """
+    Minimize a local region given selected idxs.
```

Comment:
Should this be configurable to use either L-BFGS or FIRE?

Reply:
I don't know the optimal initial step sizes for FIRE when dealing with over stretched bonds (and hoping that line search from BFGS takes care of this)